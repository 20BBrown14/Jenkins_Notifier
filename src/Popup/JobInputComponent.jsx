import React from 'react';
import InputField from 'terra-form-input/lib/InputField';
import PropTypes from 'prop-types';

const propTypes = {
/** Error message */
  errorMessage: PropTypes.string.isRequired,
  /** Whether form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for validation */
  handleValidation: PropTypes.func.isRequired,
};

const defaultProps = {
  formInvalid: false,
};

const JobInputComponent = (props) => {
  const {
    errorMessage,
    formInvalid,
    handleValidation,
  } = props;

  return (
    <InputField
      inputId="jobNameInput"
      label="Input url of job to track"
      error={errorMessage}
      help="With or without 'api/xml' appended"
      isInvalid={formInvalid}
      onChange={handleValidation}
    />
  );
};

JobInputComponent.propTypes = propTypes;
JobInputComponent.defaultProps = defaultProps;

export default JobInputComponent;
