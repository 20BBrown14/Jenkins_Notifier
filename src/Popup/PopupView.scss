.popupview-Window {
  width: 650px;
  height: 450px;
  background-color: #fff991
}