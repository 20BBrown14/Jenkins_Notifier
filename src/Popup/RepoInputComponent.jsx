import React from 'react';
import InputField from 'terra-form-input/lib/InputField';
import PropTypes from 'prop-types';


const propTypes = {
/** Error message */
  errorMessage: PropTypes.string.isRequired,
  /** Whether form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for when repo url input field changes */
  URLFieldChange: PropTypes.func.isRequired,
  /** Function for when repo name input field changes */
  nameFieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the repo URL input field */
  repoURL: PropTypes.string.isRequired,
  /** Value of the repo name input field */
  repoName: PropTypes.string.isRequired,
};

const defaultProps = {
  formInvalid: false,
};

const RepoInputComponent = (props) => {
  const {
    errorMessage,
    formInvalid,
    URLFieldChange,
    helpMessage,
    repoURL,
    repoName,
    nameFieldChanged,
  } = props;

  return (
    <div>
      <InputField
        style={{ margin: '15px 25px 12.5px 25px' }}
        inputId="repoNameInput"
        label="Name"
        value={repoName}
        onChange={nameFieldChanged}
      />
      <InputField
        style={{ margin: '12.5px 25px 12.5px 25px' }}
        inputId="repoURLInput"
        label="URL"
        error={errorMessage}
        help={helpMessage}
        isInvalid={formInvalid}
        onChange={URLFieldChange}
        value={repoURL}
      />
    </div>
  );
};

RepoInputComponent.propTypes = propTypes;
RepoInputComponent.defaultProps = defaultProps;

export default RepoInputComponent;
