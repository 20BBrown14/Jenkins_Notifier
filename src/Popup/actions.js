
/** @const action for validating repo url in state */
export const A_VALIDATE_REPO_URL = 'validate_repo_url';

/** @const action for when job input field data changed */
export const A_JOB_INPUT_DATA_CHANGED = 'job_input_data_changed';

/**
 * Action creator for A_JOB_INPUT_DATA_CHANGED
 */
export const jobInputDataChanged = newValue => (
  {
    type: A_JOB_INPUT_DATA_CHANGED,
    data: { newValue },
  }
);
