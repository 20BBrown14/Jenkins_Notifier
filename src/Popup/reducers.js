import {
  A_URL_INPUT_DATA_CHANGED,
  A_VALID_REPO_URL,
  A_INVALID_REPO_URL,
  A_NAME_INPUT_DATA_CHANGED,
  A_CONFIRM_BUTTON_CLICKED,
} from './actions';

export const POPUP_STATE = 'popupState';

const INITIAL_STATE = {
  repoURL: '',
  repoName: '',
  helpMessage: 'Repo URL should be appended with \'/api/json\'',
  formInvalid: false,
  errorMessage: undefined,
  validated: false,
  confirmed: false,
};

const reduceURLInputDataChanged = (state, action) => (
  {
    ...state,
    repoURL: action.data.newValue,
    formInvalid: false,
    helpMessage: INITIAL_STATE.helpMessage,
    errorMessage: undefined,
    validated: false,
    confirmed: false,
  }
);

const reduceNameInputDataChanged = (state, action) => (
  {
    ...state,
    repoName: action.data.newValue,
    confirmed: false,
  }
);

const reduceValidRepoURL = (state, action) => (
  {
    ...state,
    jsonData: action.data.jsonData,
    helpMessage: 'Jenkins URL is valid!',
    validated: true,
  }
);

const reduceInvalidRepoURL = (state, action) => (
  {
    ...state,
    formInvalid: true,
    errorMessage: action.data.errorMessage,
    validated: false,
  }
);

/**
 * Define Popup reducers
 * @param {object} state - State with default value of INITIAL_STATE
 * @param {string} state.repoURL - Current value of the repoURL field.
 * @param {string} state.repoName - Current value of the repoName field.Date
 * @param {string} state.helpMessage - Help message for repoURL field
 * @param {bool} state.formInvalid - Whether or not the form is invalid
 * @param {string} state.errorMessage - Error message for an invalid repo url
 * @param {bool} state.validated - Whether or not the repoURL has been validated
 * @param {bool} state.confirmed - Whether or not the confirm button has been clicked
 * @param {object} action - The action being performed
 * @param {string} action.type - The type of action being performed
 * @param {object} action.data - The data related to the action
 * @returns {object} updated state. If no action is handled returns original state.
 */
const reducePopup = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case A_URL_INPUT_DATA_CHANGED:
      return reduceURLInputDataChanged(state, action);
    case A_NAME_INPUT_DATA_CHANGED:
      return reduceNameInputDataChanged(state, action);
    case A_VALID_REPO_URL:
      return reduceValidRepoURL(state, action);
    case A_INVALID_REPO_URL:
      return reduceInvalidRepoURL(state, action);
    case A_CONFIRM_BUTTON_CLICKED:
      return { ...state, confirmed: true };
    default:
      return state;
  }
};

export default { [POPUP_STATE]: reducePopup };
