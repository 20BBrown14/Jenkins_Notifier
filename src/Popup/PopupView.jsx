import React from 'react';
import PropTypes from 'prop-types';
import Button from 'terra-button';
import Text from 'terra-text';
import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
import RepoInputComponent from './RepoInputComponent';
import './PopupView.scss';

const propTypes = {
  /** Error message for input */
  errorMessage: PropTypes.string.isRequired,
  /** If input is invalid */
  formInvalid: PropTypes.bool,
  /** Function for when repo url input field changes */
  URLFieldChange: PropTypes.func.isRequired,
  /** Function for when repo name input field changes */
  nameFieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the repo URL input field */
  repoURL: PropTypes.string.isRequired,
  /** Value of the repo name input field */
  repoName: PropTypes.string.isRequired,
  /** Handler for when validate is clicked */
  validateClickHandler: PropTypes.func,
  /** Handler for when confirm is clicked */
  confirmClickHandler: PropTypes.func,
  /** Whether or not the URL has been validated */
  validated: PropTypes.bool,
  /** Whether or not the URL has been confirmed */
  confirmed: PropTypes.bool,
};

const defaultProps = {
  formInvalid: false,
};

const PopupView = (props) => {
  const {
    errorMessage,
    formInvalid,
    URLFieldChange,
    nameFieldChanged,
    helpMessage,
    repoURL,
    validateClickHandler,
    confirmClickHandler,
    repoName,
    validated,
    confirmed,
  } = props;

  return (
    <div className="popupview-Window">
      <Text style={{ margin: '12px 0px 0px 25px' }} fontSize={32}>Track a New Repo</Text>
      <RepoInputComponent
        errorMessage={errorMessage}
        formInvalid={formInvalid}
        URLFieldChange={URLFieldChange}
        nameFieldChanged={nameFieldChanged}
        helpMessage={helpMessage}
        repoURL={repoURL}
        repoName={repoName}
      />
      <Button
        style={{ margin: '0px 5px 0px 25px' }}
        text="Validate"
        onClick={validateClickHandler}
        isDisabled={validated}
      />
      <Button
        style={{ margin: '0px 5px 0px 5px' }}
        text="Confirm"
        onClick={confirmClickHandler}
        isDisabled={confirmed}
      />
      {confirmed && validated &&
        <IconSuccess />}
    </div>
  );
};

PopupView.propTypes = propTypes;
PopupView.defaultProps = defaultProps;

export default PopupView;
