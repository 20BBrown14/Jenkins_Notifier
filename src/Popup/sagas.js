import { put, call, takeLatest } from 'redux-saga/effects';
import { A_VALIDATE_REPO_URL, A_INVALID_REPO_URL, A_VALID_REPO_URL } from './actions';
import request from '../modules/dataLoadSagas';

/**
 * Saga to validate a repo url
 * @param {object} action
 */
export function* validateRepo(action) {
  try {
    const json = yield call(request, action.data.URL);
    if (json.status === 1) {
      yield put({ type: A_VALID_REPO_URL, data: json });
    } else {
      yield put({ type: A_INVALID_REPO_URL });
    }
  } catch (e) {
    yield put({ type: A_INVALID_REPO_URL });
  }
}

function* PopupSaga() {
  yield takeLatest(A_VALIDATE_REPO_URL, validateRepo);
}

export default [PopupSaga];
