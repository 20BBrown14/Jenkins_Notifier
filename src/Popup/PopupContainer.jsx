import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import PopupView from './PopupView';

const propTypes = {
  /** If form input is invalid */
  formInvalid: PropTypes.bool,
  /** Function if input validation failed */
  inputValidationFailed: PropTypes.func.isRequired,
};

/* eslint-disable-next-line react/prefer-stateless-function */
export class PopupContainer extends React.Component {
  render() {
    const {
      formInvalid,
      fieldValidationFailed,
    } = this.props;

    const validateInput = (event) => {
      // TODO: Validate Field
    };

    return (
      <PopupView
        errorMessage="Job url is not valid. Recheck and retry."
        isInvalid={formInvalid || false}
        handleValidation={validateInput}
      />
    );
  }
}

const mapStateToProps = state => ({
  // formInvalid: TODO: Get state
});

export const mapDispatchToProps = dispatch => ({
  inputValidationFailed: (error) => {
    // TODO: Dispatch error action
  },
});

PopupContainer.propTypes = propTypes;

export default connect(mapStateToProps, mapDispatchToProps)(PopupContainer);
