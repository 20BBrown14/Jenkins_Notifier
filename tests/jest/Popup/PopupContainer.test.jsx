import React from 'react';
import { PopupContainer } from '../../../src/Popup/PopupContainer';

describe('PopupContainer', ()  => {
    describe('Initialization', () => {
        it('should render a default view', () => {
            const testContainer = (shallow(<PopupContainer inputValidationFailed={ () => {} } />));
            expect(testContainer).toMatchSnapshot();
        });
    });
});