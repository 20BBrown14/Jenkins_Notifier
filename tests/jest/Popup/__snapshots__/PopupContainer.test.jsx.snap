// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupContainer Initialization should render a default view 1`] = `
<PopupView
  URLFieldChange={[Function]}
  errorMessage="Job url is NOT valid. Recheck and retry."
  formInvalid={false}
  helpMessage=""
  nameFieldChanged={[Function]}
  repoName=""
  repoURL=""
  validateClickHandler={[Function]}
/>
`;
